README.md

```diff
@@ -287,6 +291,49 @@ Content-Type: application/pkcs7-mime
 
 ```
 
+### Use Case: Signed self extracting binary
+
+Lets create a self extracting binary by extending the example mentioned above with a signed package. In order to do this, we need to add a trailer for Package-Length in order to allow the runtime to find the start of the package. In addition for signing purposes we must add and Executable-Signature header to the package. By keeping our Executable-Signature separate from our Package-Signature it is possible to update a runtime without completely downloading a new package.
+
+Important notes:
+
+1. Package-Length must be in the Trailers for the package.
+2. Executable-Signature must use the same certificate as Package-Signature.
+3. Executable-Signature must be validated within nested packages but may not be the same as outer packages if they intend to be extracted.
+
+```html
+... binary with ability to self extract trailing webpackage ...
+Package-Signature: Li9vy3DqF8tnTXuiaAJuML3ky+er10rcgNR/VqsVpcw+ThHmYcwiB1pbOxEbzJr7; algorithm=ecdsa-with-sha384;
+Executable-Signature: Mj9wz3ErG8uoUYvjbBKvNM3lz+fs10sdhOS/WrtWqdx+UiInZdxjC1qcPyFczKs7; algorithm=ecdsa-with-sha384; certificate=cid:f47ac10b-58cc-4372-a567-0e02b2c3d479certificate=cid:f47ac10b-58cc-4372-a567-0e02b2c3d479
```

Comment:
Dumb question: why do you want an executable signature in the package? OSes wouldn't generally know to check it before running the executable, and once it's running, it can also ignore its own signature.